src/components/PostEditor/PostEditor.js

```diff
@@ -851,7 +1079,15 @@ class PostEditor extends React.Component {
             valid={valid}
             loading={loading}
             submitButtonLabel={this.buttonLabel()}
-            save={() => this.save()}
+            save={() => {
+              if (isProposal && !isEqual(proposalOptions, this.props.post.proposalOptions?.items)) {
+                if (window.confirm(t('Changing proposal options will reset the votes. Are you sure you want to continue?'))) {
```

Comment:
To discuss: if this happens we would probably need to send a notification to every person who voted, something to think about adding to the back end